pkg/expr/expr_test.go

```diff
@@ -0,0 +1,195 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package expr
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	config "istio.io/api/mixer/v1/config/descriptor"
+	dpb "istio.io/api/mixer/v1/config/descriptor"
+)
+
+func TestGoodParse(tst *testing.T) {
+	tests := []struct {
+		src     string
+		postfix string
+	}{
+		{"a.b.c == 2", "EQ($a.b.c, 2)"},
+		{`!(a.b || b || "c a b" || ( a && b ))`, `NOT(LOR(LOR(LOR($a.b, $b), "c a b"), LAND($a, $b)))`},
```

Comment:
function names have to be valid identifiers. So that excludes any operators as function names. 
I would like to minimize number of operators beyond the customary ones that we must support.